### ISSUE TITLE
Unexpected exception.

### ISSUE THREAD

**Issue by sdanyliv** [agileobjects/ReadableExpressions#100]
I think it is solvable without posting whole generated query. And yes expression contains variables which are not yet to final expression.

```
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at AgileObjects.ReadableExpressions.ExpressionAnalysis.GetUnnamedVariableNumberOrNull(ParameterExpression variable)
   at AgileObjects.ReadableExpressions.Translations.ParameterTranslation.UnnamedParameterTranslation.GetParameterName(ParameterExpression parameter, ITranslationContext context)
   at AgileObjects.ReadableExpressions.Translations.ParameterTranslation.For(ParameterExpression parameter, ITranslationContext context)
   at AgileObjects.ReadableExpressions.Translations.ExpressionTranslation.GetTranslationFor(Expression expression)
   at AgileObjects.ReadableExpressions.Translations.ExpressionTranslation.GetTranslation()
   at AgileObjects.ReadableExpressions.ExpressionExtensions.ToReadableString(Expression expression, Func`2 configuration)
   at AgileObjects.ReadableExpressions.Visualizers.ObjectSource.ExpressionVisualizerObjectSource.GetTranslationFor(Object target)
   at AgileObjects.ReadableExpressions.Visualizers.ObjectSource.ExpressionVisualizerObjectSource.GetData(Object target, Stream outgoingData, Action`2 serializer)
   at AgileObjects.ReadableExpressions.Visualizers.Vs17ExpressionVisualizerObjectSource.GetData(Object target, Stream outgoingData)
   at Microsoft.VisualStudio.DebuggerVisualizers.DebuggeeSide.Impl.ClrCustomVisualizerDebuggeeHost.GetData(Object visualizedObject)
```
Another observation, after such exceptions, Extension become unusable. And I have to restart Visual Studio.

Version 4.1.1

**Comment by SteveWilkes** [comment]
Hi!

Thanks for letting me know about this - it's an interesting one. Could you copy/paste the expression debug view for me? The error indicates it's trying to get a number suffix to ensure a unique variable name, but the code doesn't know about the `ParameterExpression` which represents the variable, which is very odd.

Cheers!

Steve

**Comment by sdanyliv** [comment]
Try this simple example. Crashes instantly.
```cs
var variable = Expression.Variable(typeof(string));
```

**Comment by SteveWilkes** [comment]
Oh, ok. Unnamed variables are translated fine within expressions, but I can see why it would fall over translating a root, unnamed variable. I'll sort it out!

**Comment by sdanyliv** [comment]
Also, please catch exceptions, to avoid restarting VS.

**Comment by SteveWilkes** [comment]
This is fixed in the latest code - and exceptions are caught and reported! - release to follow.

Thanks!
Steve

**Comment by SteveWilkes** [comment]
This is fixed in v3.1.1 of [the NuGet package](https://www.nuget.org/packages/AgileObjects.ReadableExpressions), and v4.2 of [the visualizers](https://marketplace.visualstudio.com/items?itemName=vs-publisher-1232914.ReadableExpressionsVisualizers). Cheers!
